test/datafmt.jl

```diff
@@ -93,6 +93,14 @@ let x = [1,2,3], y = [4,5,6], io = IOBuffer()
     @test readcsv(io) == [x y]
 end
 
+let x = [0.1 0.3 0.5], io = IOBuffer()
```

Comment:
Using `readdlm` without the correct delimiter doesn't make much sense to me. I'd simply check that the contents of the buffer are what you expect using `readall`, and then use `readcsv` as in the previous test to check that we can parse the file correctly.